erpnext/manufacturing/doctype/production_plan/production_plan.py

```diff
@@ -165,10 +167,24 @@ def get_mr_items(self):
 		self.calculate_total_planned_qty()
 
 	def add_items(self, items):
-		self.set('po_items', [])
+		refs = {}
 		for data in items:
 			item_details = get_item_details(data.item_code)
+			if self.combine_items:	
+				if item_details.bom_no in refs.keys():
+					refs[item_details.bom_no]['qty'] += data.pending_qty
+					refs[item_details.bom_no]['so'].append(data.parent)	
+					refs[item_details.bom_no]['so_items'].append(data.name)
+					refs[item_details.bom_no]['planned_qty'].append(data.pending_qty)
+					continue
+				else:
+					refs[item_details.bom_no] = {'qty': data.pending_qty, 'ref': data.name}
+					refs[item_details.bom_no]['so'] = [data.parent]
+					refs[item_details.bom_no]['so_items'] = [data.name]
```

Comment:
data.name used in line no 181 as well on 183, better you create a structure with tuple like as below

```
key = ('sales_order', 'bom_no')

# data.name is the reference of po_items
refs = {key: ['data.name', 'data1.name']}
```